text/0000-arrow-deref.md

```diff
@@ -0,0 +1,135 @@
+1. Accepting the `RArrow` token in tandem with the `Dot` token, in `parse_expr_dot_or_call_with_`. 
+2. Renaming `parse_dot_suffix` to `parse_dot_or_arrow_suffix` and passing a unary `UnOp::Deref` expression in `self_arg`.
+
+There are no grammar ambiguities with respect to other instances of `RArrow`, such as `Fn() -> T`.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The RArrow token could have other future use cases.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The proposal makes unsafe code, that which is the most safety-critical code, easier to read, understand, and maintain.
+
+It also prevents preferring references over raw pointers. This prevents common mistakes that create UB by simultaneous mutable references.
+
+As discussed in the article [Rust's Unsafe Pointer Types Need An Overhaul](https://faultlore.com/blah/fix-rust-pointers/), the Tilde token could be used for walking field pointers of different types without changing the level of indirection. The proposed arrow operator is different.
+The arrow dereferences and yields a place expression. This is important because it is the only way to completely eliminate excess parentheses. Suppose we used the Tilde token for obtaining pointers to fields. Then the example before would be written as:
+
+```rust
+(*(**pointer.add(5)~some_field).method_returning_pointer()).other_method()
+```
+
+Which does not solve the problem.
+
+# Prior art
+[prior-art]: #prior-art
+
+This exists in both C and C++. If for example C had neither the arrow nor an alternative auto-deref, writing C code would be quite cumbersome. Early development of Zig also incorporated the utility of non-prefix operators for dereference and auto-deref. Like 
+[Rust itself](https://github.com/rust-lang/rfcs/pull/102). However, Rust makes a clear distinction between unsafe and safe, where we need to be cautious.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+One unresolved question is whether the arrow operator should only be capable of dereferencing raw pointers. Should it be a sugar that desugars to `(*x).y`, no matter the type of `x`, or should `x` only be allowed to be a raw pointer? Implementing this requires an extra type-checking step in addition to the `rustc_parse` implementation outlined above.
+
+Maybe a trait called UnsafeDeref? This way it can be implemented for both raw pointers and NonNull<T>.
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+There have been many discussions over the years.
+
+[Need for -> operator for Unsafe Code Guidelines](https://internals.rust-lang.org/t/need-for-operator-for-unsafe-code-guidelines/10022)
+
+[Add arrow operator as sugar for (*var)](https://github.com/gpuweb/gpuweb/issues/4114)
+
+[Rust's Unsafe Pointer Types Need An Overhaul](https://faultlore.com/blah/fix-rust-pointers/)
```

Comment:
How are these "future possibilities"? These are "prior art", i.e. past discussions of the same / a similar proposal. This RFC should summarize those discussions and explain how it resolves the concerns that were brought up in the past.